### ISSUE TITLE
conda-smithy seems to be incompatible with conda 4.7.6 / conda-build 3.18.7

### ISSUE THREAD

**Issue by xhochy** [conda-forge/conda-smithy#1105]
Updating to the latest combination of both a `conda smithy rerender` in https://github.com/conda-forge/jupyterlab_code_formatter-feedstock fails with:

```
% conda smithy rerender
Traceback (most recent call last):
  File "/Users/uwe/miniconda3/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda_smithy/cli.py", line 485, in main
    args.subcommand_func(args)
  File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda_smithy/cli.py", line 353, in __call__
    check=args.check
  File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda_smithy/configure_feedstock.py", line 1404, in main
    r = conda_build.conda_interface.Resolve(index)
  File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 116, in __init__
    self._system_precs = {_ for _ in index if _.package_type == PackageType.VIRTUAL_SYSTEM}
  File "/Users/uwe/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 116, in <setcomp>
    self._system_precs = {_ for _ in index if _.package_type == PackageType.VIRTUAL_SYSTEM}
AttributeError: 'Dist' object has no attribute 'package_type
```

Environment (<code>conda list</code>):
<details>

```
# packages in environment at /Users/uwe/miniconda3:
#
# Name                    Version                   Build  Channel
appdirs                   1.4.3                      py_1    conda-forge
asn1crypto                0.24.0                py37_1003    conda-forge
astroid                   2.2.5                    py37_0    conda-forge
beautifulsoup4            4.7.1                 py37_1001    conda-forge
blinker                   1.4                        py_1    conda-forge
bzip2                     1.0.6             h1de35cc_1002    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_0    conda-forge
cffi                      1.12.3           py37hccf1714_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
click                     7.0                        py_0    conda-forge
conda                     4.7.6                    py37_0    conda-forge
conda-build               3.18.7                   py37_0    conda-forge
conda-env                 2.6.0                         1    conda-forge
conda-forge-pinning       2019.07.10                    0    conda-forge
conda-pack                0.3.0+3.gec86459           dev_0    <develop>
conda-package-handling    1.3.11                   py37_0    conda-forge
conda-smithy              3.4.0                      py_1    conda-forge
conda-verify              3.1.1                 py37_1000    conda-forge
cryptography              2.7              py37h212c5bf_0    conda-forge
curl                      7.64.1               h22ea746_0    conda-forge
deprecated                1.2.5                      py_0    conda-forge
docker-py                 4.0.2                    py37_0    conda-forge
docker-pycreds            0.4.0                      py_0    conda-forge
expat                     2.2.5             h6de7cb9_1003    conda-forge
filelock                  3.0.10                     py_0    conda-forge
future                    0.17.1                py37_1000    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
git                       2.22.0          pl526hbdf3604_0    conda-forge
gitdb2                    2.0.5                      py_0    conda-forge
gitpython                 2.1.11                     py_0    conda-forge
glob2                     0.7                        py_0    conda-forge
gunicorn                  19.9.0                py37_1000    conda-forge
icu                       58.2              h0a44026_1000    conda-forge
idna                      2.8                   py37_1000    conda-forge
isodate                   0.6.0                      py_1    conda-forge
isort                     4.3.21                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
krb5                      1.16.3            hcfa6398_1001    conda-forge
lazy-object-proxy         1.4.1            py37h01d97ff_0    conda-forge
libarchive                3.3.3             h5c473cc_1006    conda-forge
libcurl                   7.64.1               h16faf7d_0    conda-forge
libcxx                    8.0.0                         4    conda-forge
libcxxabi                 8.0.0                         4    conda-forge
libedit                   3.1.20170329      hcfe32e1_1001    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblief                   0.9.0                h2a1bed3_1    conda-forge
libssh2                   1.8.2                hcdc9a53_2    conda-forge
libxml2                   2.9.9                hd80cff7_1    conda-forge
lz4                       2.1.9            py37h9608888_0    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
lzo                       2.10              h1de35cc_1000    conda-forge
markupsafe                1.1.1            py37h1de35cc_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
msrest                    0.6.8                      py_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
oauthlib                  3.0.1                      py_0    conda-forge
openssl                   1.1.1c               h01d97ff_0    conda-forge
packaging                 19.0                       py_0    conda-forge
perl                      5.26.2            haec8ef5_1006    conda-forge
pip                       19.1.1                   py37_0    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
psutil                    5.6.3            py37h01d97ff_0    conda-forge
py-lief                   0.9.0            py37h6d6d4d2_1    conda-forge
pycosat                   0.6.3           py37h1de35cc_1001    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pycrypto                  2.6.1           py37h1de35cc_1002    conda-forge
pygithub                  1.43.6                   py37_0    conda-forge
pyjwt                     1.7.1                      py_0    conda-forge
pylint                    2.3.1                    py37_0    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pyparsing                 2.4.0                      py_0    conda-forge
pysocks                   1.7.0                    py37_0    conda-forge
python                    3.7.3                h93065d6_1    conda-forge
python-libarchive-c       2.8                   py37_1004    conda-forge
pytz                      2019.1                     py_0    conda-forge
pyyaml                    5.1.1            py37h01d97ff_0    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
requests                  2.22.0                   py37_0    conda-forge
requests-oauthlib         1.2.0                      py_0    conda-forge
ripgrep                   11.0.1                        0    conda-forge
ruamel                    1.0                      py37_0    conda-forge
ruamel.yaml               0.15.98          py37h01d97ff_0    conda-forge
ruamel_yaml               0.15.71         py37h1de35cc_1000    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
shyaml                    0.6.0                      py_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
soupsieve                 1.9.2                    py37_0    conda-forge
sqlite                    3.29.0               hb7d70f7_0    conda-forge
tk                        8.6.9             h2573ce8_1002    conda-forge
tqdm                      4.32.2                     py_0    conda-forge
urllib3                   1.24.3                   py37_0    conda-forge
vsts-python-api           0.1.22                     py_0    conda-forge
websocket-client          0.56.0                   py37_0    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
wrapt                     1.11.2           py37h01d97ff_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
yaml                      0.1.7             h1de35cc_1001    conda-forge
zlib                      1.2.11            h1de35cc_1004    conda-forge
zstd                      1.4.0                ha9f0a20_0    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
     active environment : base
    active env location : /Users/uwe/miniconda3
            shell level : 1
       user config file : /Users/uwe/.condarc
 populated config files : /Users/uwe/.condarc
          conda version : 4.7.6
    conda-build version : 3.18.7
         python version : 3.7.3.final.0
       virtual packages :
       base environment : /Users/uwe/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/uwe/miniconda3/pkgs
                          /Users/uwe/.conda/pkgs
       envs directories : /Users/uwe/miniconda3/envs
                          /Users/uwe/.conda/envs
               platform : osx-64
             user-agent : conda/4.7.6 requests/2.22.0 CPython/3.7.3 Darwin/18.6.0 OSX/10.14.5
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</details>

**Comment by xhochy** [comment]
Same error as in https://github.com/conda-forge/staged-recipes/issues/8798

**Comment by Luthaf** [comment]
I had the same issue with conda-smithy 3.4.0, but version 3.4.1 fixed it for me (with conda 4.7.8 and conda-build 3.18.7)

**Comment by xhochy** [comment]
This has been resolved
